app/uk/gov/hmrc/exports/services/mapping/declaration/TypeCodeBuilder.scala

```diff
@@ -36,11 +38,18 @@ class TypeCodeBuilder @Inject()() extends ModifyingBuilder[ExportsDeclaration, D
     declaration.setTypeCode(typeCodeType)
   }
 
-  private def createTypeCode(decType: AdditionalDeclarationType, dispatchLocation: Option[DispatchLocation]): DeclarationTypeCodeType = {
+  private def createTypeCode(decType: AdditionalDeclarationType, dispatchLocation: String): DeclarationTypeCodeType = {
     val typeCodeType = new DeclarationTypeCodeType()
-    dispatchLocation.foreach { data =>
-      typeCodeType.setValue(data.dispatchLocation + decType.toString)
-    }
+    typeCodeType.setValue(dispatchLocation + decType.toString)
     typeCodeType
   }
+
+  private val countriesFor_CO_declType = List("GG", "JE", "Jersey", "Guernsey")
```

Comment:
Why is there both country name and country code? We should be using either codes, or names.

Reply:
Because I was not sure what values `locations.destinationCountry.code` can have, if codes only or countries only or both. If you know what are the possible values I have no problem of course to only include those.

Reply:
We only store country codes. There was a confusion some time ago, because it turned out we store it inconsistently, but it is not resolved.

Reply:
:+1:

Reply:
I removed "Jersey" and "Guernsey" from the list and that's ok, but now I'm thinking that there was no harm in leaving in the list and we would have been future-proof. At the end what is relevant is if the destination country is "Jersey" or "Guernsey", by the code or by the name is not important.